### ISSUE TITLE
Lambda cannot get_function if created with a ZipFile

### ISSUE THREAD

**Issue by oharaandrew314** [spulec/moto#1136]
If you create a lambda with `create_function` while providing the code with the `ZipFile` argument, then getting the function with `get_function` will result in an error:

```
def get_code(self):
        return {
            "Code": {
                "Location": "s3://lambda-functions.aws.amazon.com/{0}".format(
>                   self.code['S3Key']),
                "RepositoryType": "S3"
            },
            "Configuration": self.get_configuration(),
        }
E       TypeError: byte indices must be integers or slices, not str
```

To reproduce, in`tests/test_awslambda/test_awslambda.py:test_create_function_from_zipfile`, just add the line `conn.get_function(FunctionName='testFunction')` to the end of the test and you'll get the error.

This should be fixed.  I am willing to look into this myself a little bit.

**Comment by dbfr3qs** [comment]
I have submitted a fix for this [here](https://github.com/spulec/moto/pull/1144).

**Comment by dbfr3qs** [comment]
This is now fixed in Moto version 1.1.8.

**Comment by oharaandrew314** [comment]
Judging by the unit test, it looks like that should fix it.  Thanks!
